mg: new recipe

hoanga opened the pull request:
the following recipe adds the mg editor

sample post-install below:

```
> uname -a
Haiku shredder 1 hrev56085 Apr 29 2022 07:35:01 x86_64 x86_64 Haiku

> man mg | head
MG(1)                       General Commands Manual                      MG(1)

NAME
     mg - emacs-like text editor

SYNOPSIS
     mg [-nR] [-b file] [-f mode] [-u file] [+number] [file ...]

DESCRIPTION
     mg is intended to be a small, fast, and portable editor for people who

> which mg
/bin/mg
```

Begasus commented on app-editors/mg/mg-7.0.recipe:
```diff
@@ -0,0 +1,47 @@
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+        devel:libncursesw$secondaryArchSuffix
```
1 tab here, no spaces

Begasus commented on app-editors/mg/mg-7.0.recipe:
```diff
@@ -0,0 +1,47 @@
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+        devel:libncursesw$secondaryArchSuffix
+	cmd:make
```
Don't think make is required for the devel package (already in BUILD_PREREQUIRES)
```diff
- 	cmd:make
```

Begasus commented on app-editors/mg/mg-7.0.recipe:
```diff
@@ -0,0 +1,47 @@
+
+BUILD()
+{
+	runConfigure ./configure --prefix="$prefix"
+	make
```
No need for prefix, runConfigure takes care of that:
```diff
- 	runConfigure ./configure --prefix="$prefix"
-  	make
+ 	runConfigure ./configure
+ 	make $jobArgs
```

Begasus replied:
Didn't check a build yet, but I guess you could use $jobArgs here :)

Begasus replied:
Only prefix left, will do a build here to check :)

Begasus commented on app-editors/mg/mg-7.0.recipe:
```diff
@@ -0,0 +1,46 @@
+people who can't (or don't want to) run emacs for one reason or another, \
+or are not familiar with the vi editor."
+HOMEPAGE="https://github.com/ibara/mg"
+COPYRIGHT="2000"
```
Maybe:
```diff
- COPYRIGHT="2000"
+ COPYRIGHT="2019-2021 Brian Callahan"
```

Begasus commented on app-editors/mg/mg-7.0.recipe:
```diff
@@ -0,0 +1,46 @@
+
+PROVIDES="
+	mg$secondaryArchSuffix = $portVersion
+	cmd:mg$secondaryArchSuffix = $portVersion
```
cmd is installed in $prefix/bin, so:
```diff
- 	cmd:mg$secondaryArchSuffix = $portVersion
+ 	cmd:mg = $portVersion
```

Begasus commented on app-editors/mg/mg-7.0.recipe:
```diff
@@ -0,0 +1,46 @@
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= 1"
```
Probably a leftover?
```diff
-libVersion="$portVersion"
-libVersionCompat="$libVersion compat >= 1"
```

Begasus commented:
After looking at some recipes through repology, I see a lot of them using https://github.com/hboetes/mg for the sources
Even Gentoo does: https://packages.gentoo.org/packages/app-editors/mg

hoanga commented:
thanks for the quick feedback and i had a brief look at the hboetes version of mg and it seems that it uses a different approach for the port and would require another library (libbsd) to be available. the tagging scheme for each of the other cited mg ports are also different (hboetes uses timestamp, troglobit follows semver but at a different version).

for the time being, i have pushed an update that addresses suggestions in case this flavor of the mg port is chosen.

Begasus commented:
Thanks!
